docs/additional/interesting_aspects_en.md

```diff
@@ -0,0 +1,249 @@
+More information on multi-stage build is available on the [official Docker website][docker-multi-stage].
+
+[docker-multi-stage]: https://docs.docker.com/develop/develop-images/multistage-build/
+
+### CES-free development environment.
+
+Building the Dogu and delivering it to the local EcoSystem not only takes a lot of time, but also takes away the ability to debug the software effectively.  
+Therefore, it makes sense to build a local development environment independent of the CES to effectively run, test and debug the Dogu software.
+
+The local development environment must provide all dependent CES services through locally started alternatives.
+We recommend that a local development environment is defined in the form of a `docker-compose.yml` file and can be easily started at any time with `docker-compose up` and shut down with `docker-compose down`.  
+For an example [see CAS][local-cas-example].
+
+[local-cas-example]: https://github.com/cloudogu/cas/blob/develop/app/docker-compose.yml
+
+## Quality Assurance for Dogus
+
+### Container Validation (GOSS)
+
+Container validation assures the correct configuration of the container, which is necessary for the smooth execution of the software.  
+
+For this purpose, we use the server validation program [goss][goss] in our Dogus.  
+The actual goss tests are created in the following data structure in the dogu:
+- `<root directory of the Dogu>/`
+  - `spec/`
+    - `goss/`
+      - `goss.yml` (contains all goss tests)
+
+The goss tests should ideally check all important aspects for the smooth start of a dogu.  
+This includes aspects like:
+- Checking permission for scripts (startup.sh), volumes.
+- Checking the UID & GID for files (startup.sh, resources, volumes, write-dependent paths)
+- Accessibility of the software: TCP health checks
+
+For pipeline integration, we recommend using the `ecoSystem.verify()` method from the [dogu-build-lib][dogu-build-lib].  
+An example call looks like the following in a Jenkinsfile:
+
+```groovy
+stage('Verify') {
+    ecoSystem.verify("/dogu")
+}
+```
+
+This internally calls the `cesapp verify` command, which runs the goss tests.
+
+[goss]: https://github.com/goss-org/goss
+[dogu-build-lib]: https://github.com/cloudogu/dogu-build-lib
+
+## Documentation
+
+Every tool should have a reasonable documentation about used features, which supports usage or support.
+To allow easy integration into our existing documentation infrastructure, the documentation should be placed in a `/docs` folder located in the repository root.
+
+By default, the docs include the following:
+- Features of the Dogu
+- Installation of the Dogu
+- Configuration of the Dogu
+- Development on the Dogu
+- System requirements of the Dogu (nice to have)
+- Readme
+- Changelog
+
+### License
+
+The license for using the Dogu should be clearly stated.
+This can be specified in the Readme, for example.
```

Comment:
It would also help to provide a full license file if one exists, f.i. in case of a publicly available repository (it does not have to be a FOSS repository)